### PR TITLE
add color to labels

### LINKED ISSUE #227: Add color property to MinutesLabel

### PR DESCRIPTION
**Pull request by Nef10**
fix #227

### CODE REVIEW

File: _1327/minutes/models.py [hunk 1]
```diff
@@ -12,6 +13,31 @@
...
+class MinutesLabel(models.Model):
+	title = models.CharField(max_length=255)
+	color = HexColorModelField()
+
+	def __str__(self):
+		return self.title
+
+	@property
+	def text_color(self):
+		r = int(self.color[1:3], 16)
```
Comment (janno42) [inline _1327/minutes/models.py]: in the regex you allow `#fff` as well as `#ffffff`. what happens here if your color does only have 3 digits?

File: _1327/minutes/migrations/0007_auto_20160208_2122.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    operations = [
+        migrations.RemoveField(
+            model_name='minuteslabel',
+            name='minutes',
+        ),
+        migrations.AddField(
+            model_name='minutesdocument',
+            name='labels',
+            field=models.ManyToManyField(related_name='minutes', to='minutes.MinutesLabel', blank=True),
```
Comment (Nef10) [inline _1327/minutes/migrations/0007_auto_20160208_2122.py]: @invliD please check if you migration script needs adjustments to this

File: _1327/minutes/models.py [hunk 3]
```diff
@@ -12,6 +13,31 @@
...
+		g = int(self.color[3:5], 16)
+		b = int(self.color[5:7], 16)
+		yiq = ((r * 299) + (g * 587) + (b * 114)) / 1000
```
Comment (invliD) [inline _1327/minutes/models.py]: Please add a comment!

File: _1327/minutes/models.py [hunk 4]
```diff
@@ -12,6 +13,31 @@
...
+		return '#171717' if (yiq >= 128) else '#ebeced'
```
Comment (invliD) [inline _1327/minutes/models.py]: Wouldn't it make more sense to use bootstrap classes instead of hardcoded colors?

File: _1327/minutes/models.py [hunk 5]
```diff
@@ -12,6 +13,38 @@
...
+class HexColorModelField(models.CharField):
+
+	def __init__(self, *args, **kwargs):
+		kwargs['max_length'] = 7
+		super(HexColorModelField, self).__init__(*args, **kwargs)
+
+	def to_python(self, value):
+		" Normalize hex color to 6 digits"
+		if value[0:1] == '#' and len(value) == 4:
+			return '#' + value[1:2] + value[1:2] + value[2:3] + value[2:3] + value[3:4] + value[3:4]
```
Comment (invliD) [inline _1327/minutes/models.py]: Please use `value[1:2] * 2` etc. for string repetition.

File: _1327/minutes/models.py [hunk 6]
```diff
@@ -12,6 +13,38 @@
 MINUTES_VIEW_PERMISSION_NAME = 'view_minutesdocument'
 
 
+class HexColorModelField(models.CharField):
```
Comment (invliD) [inline _1327/minutes/models.py]: Move this class to forms.py if possible.

File: _1327/minutes/fields.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+class HexColorModelField(models.CharField):
+
+	def __init__(self, *args, **kwargs):
+		kwargs['max_length'] = 7
+		super(HexColorModelField, self).__init__(*args, **kwargs)
+
+	def to_python(self, value):
+		" Normalize hex color to 6 digits"
+		if value[0:1] == '#' and len(value) == 4:
```
Comment (Bartzi) [inline _1327/minutes/fields.py]: how about `value[0]' here?

File: _1327/minutes/fields.py [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+			return '#' + value[1:2] * 2 + value[2:3] * 2 + value[3:4] * 2
```
Comment (Bartzi) [inline _1327/minutes/fields.py]: same here `value[1]`, `value[2]` and `value[3]`

**Comment by Bartzi** [comment]
:+1:

**Pull request merged**